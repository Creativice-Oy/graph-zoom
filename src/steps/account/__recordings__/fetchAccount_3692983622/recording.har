{
  "log": {
    "_recordingName": "fetchAccount",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "32b6c540dbe7339d81f7948939897f76",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.zoom.us"
            }
          ],
          "headersSize": 862,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://api.zoom.us/v2/users/me"
        },
        "response": {
          "bodySize": 739,
          "content": {
            "mimeType": "application/json;charset=UTF-8",
            "size": 739,
            "text": "{\"id\":\"H9Yx0-O8QUCXVIseYAsjaA\",\"first_name\":\"JupiterOne\",\"last_name\":\"Development\",\"email\":\"development@jupiterone.dev\",\"type\":2,\"role_name\":\"Owner\",\"pmi\":4440405982,\"use_pmi\":false,\"personal_meeting_url\":\"https://zoom.us/j/4440405982?pwd=c0Jidml0ZS9ZRzA3dEZWSTNHNHlPZz09\",\"timezone\":\"America/New_York\",\"verified\":1,\"dept\":\"\",\"created_at\":\"2021-10-27T00:00:49Z\",\"last_login_time\":\"2021-11-02T09:51:57Z\",\"host_key\":\"815201\",\"cms_user_id\":\"\",\"jid\":\"h9yx0-o8qucxviseyasjaa@xmpp.zoom.us\",\"group_ids\":[\"ppUAdwEBRFCdRE0mfLttog\"],\"im_group_ids\":[],\"account_id\":\"N0oChUdLTrCZLfk9JXp3Hw\",\"language\":\"\",\"phone_country\":\"\",\"phone_number\":\"\",\"status\":\"active\",\"job_title\":\"\",\"location\":\"\",\"login_types\":[100],\"role_id\":\"0\",\"account_number\":7011249145}"
          },
          "cookies": [
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_aid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_haid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_tmaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_htmaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "httpOnly": true,
              "name": "cred",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "_zm_ctaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "_zm_chtaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Wed, 03 Nov 2021 04:43:44 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=UTF-8"
            },
            {
              "name": "content-length",
              "value": "739"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "x-zm-trackingid",
              "value": "v=2.0;clid=aw1;rid=WEB_6b57aea1eaef35fe8e753d8acb99267a"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "cache-control",
              "value": "no-cache, no-store, must-revalidate, no-transform"
            },
            {
              "name": "pragma",
              "value": "no-cache"
            },
            {
              "name": "expires",
              "value": "Thu, 01 Jan 1970 00:00:00 GMT"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "vary",
              "value": "Origin, Access-Control-Request-Method, Access-Control-Request-Headers"
            },
            {
              "name": "x-ratelimit-category",
              "value": "Light"
            },
            {
              "name": "x-zm-zoneid",
              "value": "VA2"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "referrer-policy",
              "value": "strict-origin-when-cross-origin"
            }
          ],
          "headersSize": 1347,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-11-03T04:43:43.081Z",
        "time": 1176,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 1176
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
