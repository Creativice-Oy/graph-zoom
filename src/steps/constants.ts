import {
  RelationshipClass,
  StepEntityMetadata,
  StepRelationshipMetadata,
} from '@jupiterone/integration-sdk-core';

export enum IntegrationSteps {
  USERS = 'fetch-users',
  GROUPS = 'fetch-groups',
  BUILD_USER_AND_GROUP_RELATIONSHIP = 'build-user-and-group-relationship',
  ROLES = 'fetch-roles',
}

export const Entities: Record<'USER' | 'GROUP' | 'ROLE', StepEntityMetadata> = {
  USER: {
    resourceName: 'User',
    _type: 'zoom_user',
    _class: 'User',
  },
  GROUP: {
    resourceName: 'Group',
    _type: 'zoom_group',
    _class: 'Group',
  },
  ROLE: {
    resourceName: 'Role',
    _type: 'zoom_role',
    _class: 'AccessRole',
  },
};

export const Relationships: Record<
  'GROUP_HAS_USER',
  StepRelationshipMetadata
> = {
  GROUP_HAS_USER: {
    _type: 'zoom_group_has_user',
    _class: RelationshipClass.HAS,
    sourceType: Entities.GROUP._type,
    targetType: Entities.USER._type,
  },
};
