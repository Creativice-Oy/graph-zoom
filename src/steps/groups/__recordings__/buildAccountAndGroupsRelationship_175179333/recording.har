{
  "log": {
    "_recordingName": "buildAccountAndGroupsRelationship",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "32b6c540dbe7339d81f7948939897f76",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.zoom.us"
            }
          ],
          "headersSize": 862,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://api.zoom.us/v2/users/me"
        },
        "response": {
          "bodySize": 739,
          "content": {
            "mimeType": "application/json;charset=UTF-8",
            "size": 739,
            "text": "{\"id\":\"H9Yx0-O8QUCXVIseYAsjaA\",\"first_name\":\"JupiterOne\",\"last_name\":\"Development\",\"email\":\"development@jupiterone.dev\",\"type\":2,\"role_name\":\"Owner\",\"pmi\":4440405982,\"use_pmi\":false,\"personal_meeting_url\":\"https://zoom.us/j/4440405982?pwd=c0Jidml0ZS9ZRzA3dEZWSTNHNHlPZz09\",\"timezone\":\"America/New_York\",\"verified\":1,\"dept\":\"\",\"created_at\":\"2021-10-27T00:00:49Z\",\"last_login_time\":\"2021-11-02T09:51:57Z\",\"host_key\":\"815201\",\"cms_user_id\":\"\",\"jid\":\"h9yx0-o8qucxviseyasjaa@xmpp.zoom.us\",\"group_ids\":[\"ppUAdwEBRFCdRE0mfLttog\"],\"im_group_ids\":[],\"account_id\":\"N0oChUdLTrCZLfk9JXp3Hw\",\"language\":\"\",\"phone_country\":\"\",\"phone_number\":\"\",\"status\":\"active\",\"job_title\":\"\",\"location\":\"\",\"login_types\":[100],\"role_id\":\"0\",\"account_number\":7011249145}"
          },
          "cookies": [
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_aid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_haid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_tmaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_htmaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "httpOnly": true,
              "name": "cred",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "_zm_ctaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "_zm_chtaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Wed, 03 Nov 2021 05:00:24 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=UTF-8"
            },
            {
              "name": "content-length",
              "value": "739"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "x-zm-trackingid",
              "value": "v=2.0;clid=aw1;rid=WEB_ff81e1283a2aa139c3bb109408d22338"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "cache-control",
              "value": "no-cache, no-store, must-revalidate, no-transform"
            },
            {
              "name": "pragma",
              "value": "no-cache"
            },
            {
              "name": "expires",
              "value": "Thu, 01 Jan 1970 00:00:00 GMT"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "vary",
              "value": "Origin, Access-Control-Request-Method, Access-Control-Request-Headers"
            },
            {
              "name": "x-ratelimit-category",
              "value": "Light"
            },
            {
              "name": "x-zm-zoneid",
              "value": "VA2"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "referrer-policy",
              "value": "strict-origin-when-cross-origin"
            }
          ],
          "headersSize": 1347,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-11-03T05:00:23.589Z",
        "time": 1730,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 1730
        }
      },
      {
        "_id": "7892a632e9fd0e489b37b35fcbd815a2",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.zoom.us"
            }
          ],
          "headersSize": 860,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://api.zoom.us/v2/groups"
        },
        "response": {
          "bodySize": 258,
          "content": {
            "mimeType": "application/json;charset=UTF-8",
            "size": 258,
            "text": "{\"total_records\":2,\"groups\":[{\"id\":\"ppUAdwEBRFCdRE0mfLttog\",\"name\":\"Test Group\",\"total_members\":2,\"in_directory\":false,\"directory_privacy\":2},{\"id\":\"dSHPJbpfQo6uS1a0_iznyg\",\"name\":\"Test Group 2\",\"total_members\":0,\"in_directory\":false,\"directory_privacy\":2}]}"
          },
          "cookies": [
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_aid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_haid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_tmaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_htmaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "httpOnly": true,
              "name": "cred",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "_zm_ctaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "_zm_chtaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Wed, 03 Nov 2021 05:00:26 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=UTF-8"
            },
            {
              "name": "content-length",
              "value": "258"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "x-zm-trackingid",
              "value": "v=2.0;clid=aw1;rid=WEB_f0a5fb42877dd0301eede101d30665f1"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "cache-control",
              "value": "no-cache, no-store, must-revalidate, no-transform"
            },
            {
              "name": "pragma",
              "value": "no-cache"
            },
            {
              "name": "expires",
              "value": "Thu, 01 Jan 1970 00:00:00 GMT"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "vary",
              "value": "Origin, Access-Control-Request-Method, Access-Control-Request-Headers"
            },
            {
              "name": "x-ratelimit-category",
              "value": "Medium"
            },
            {
              "name": "x-zm-zoneid",
              "value": "VA2"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "referrer-policy",
              "value": "strict-origin-when-cross-origin"
            }
          ],
          "headersSize": 1348,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-11-03T05:00:25.328Z",
        "time": 905,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 905
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
