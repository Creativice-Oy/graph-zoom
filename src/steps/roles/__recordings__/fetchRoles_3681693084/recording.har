{
  "log": {
    "_recordingName": "fetchRoles",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "0b6d856aa89cbe61f82ab25f5324f4ef",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "*/*"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.zoom.us"
            }
          ],
          "headersSize": 859,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://api.zoom.us/v2/roles"
        },
        "response": {
          "bodySize": 448,
          "content": {
            "mimeType": "application/json;charset=UTF-8",
            "size": 448,
            "text": "{\"total_records\":3,\"roles\":[{\"id\":\"0\",\"name\":\"Owner\",\"description\":\"Account owner has full privileges to access and manage a Zoom account.\",\"total_members\":1},{\"id\":\"1\",\"name\":\"Admin\",\"description\":\"Admins have wide range privileges to access and manage a Zoom account.\",\"total_members\":0},{\"id\":\"2\",\"name\":\"Member\",\"description\":\"Members have access to basic Zoom video meeting functions but no account management privileges.\",\"total_members\":1}]}"
          },
          "cookies": [
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_aid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_haid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_tmaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "zm_htmaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "httpOnly": true,
              "name": "cred",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "_zm_ctaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            },
            {
              "domain": ".zoom.us",
              "expires": "1970-01-01T00:00:10.000Z",
              "httpOnly": true,
              "name": "_zm_chtaid",
              "path": "/",
              "secure": true,
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "date",
              "value": "Thu, 28 Oct 2021 10:18:08 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=UTF-8"
            },
            {
              "name": "content-length",
              "value": "448"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "x-zm-trackingid",
              "value": "v=2.0;clid=aw1;rid=WEB_fdba951d4fd8cf0f457e0f8f1d593ebd"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "cache-control",
              "value": "no-cache, no-store, must-revalidate, no-transform"
            },
            {
              "name": "pragma",
              "value": "no-cache"
            },
            {
              "name": "expires",
              "value": "Thu, 01 Jan 1970 00:00:00 GMT"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "vary",
              "value": "Origin, Access-Control-Request-Method, Access-Control-Request-Headers"
            },
            {
              "name": "x-ratelimit-category",
              "value": "Medium"
            },
            {
              "name": "x-zm-zoneid",
              "value": "VA2"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "referrer-policy",
              "value": "strict-origin-when-cross-origin"
            }
          ],
          "headersSize": 1348,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-10-28T10:18:07.607Z",
        "time": 1249,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 1249
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
