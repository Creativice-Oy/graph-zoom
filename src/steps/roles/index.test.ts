import {
  createMockStepExecutionContext,
  Recording,
} from '@jupiterone/integration-sdk-testing';
import { IntegrationConfig } from '../../config';
import { buildUserAndRolesRelationship, fetchRoles } from '.';
import { integrationConfig } from '../../../test/config';
import { setupZoomRecording } from '../../../test/recording';
import { fetchUsers } from '../users';
import { Relationships } from '../constants';

describe('#fetchRoles', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupZoomRecording({
      directory: __dirname,
      name: 'fetchRoles',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchRoles(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      collectedEntities: context.jobState.collectedEntities,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    const roles = context.jobState.collectedEntities.filter((e) =>
      e._type.includes('zoom_role'),
    );
    expect(roles.length).toBeGreaterThan(0);
    expect(roles).toMatchGraphObjectSchema({
      _class: ['AccessRole'],
      schema: {
        additionalProperties: false,
        properties: {
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          _type: { const: 'zoom_role' },
          id: { type: 'string' },
          name: { type: 'string' },
          totalMembers: { type: 'number' },
          description: { type: 'string' },
        },
      },
    });
  });
});

describe('#buildUserAndRolesRelationship', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupZoomRecording({
      directory: __dirname,
      name: 'buildUserAndRolesRelationship',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should build role and user relationship', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchUsers(context);
    await fetchRoles(context);
    await buildUserAndRolesRelationship(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      collectedEntities: context.jobState.collectedEntities,
      encounteredTypes: context.jobState.encounteredTypes,
      collectedRelationships: context.jobState.collectedRelationships,
    }).toMatchSnapshot();

    const users = context.jobState.collectedEntities.filter((e) =>
      e._type.includes('zoom_user'),
    );
    expect(users.length).toBeGreaterThan(0);
    expect(users).toMatchGraphObjectSchema({
      _class: ['User'],
      schema: {
        additionalProperties: false,
        properties: {
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          _type: { const: 'zoom_user' },
          name: { type: 'string' },
          username: { type: 'string' },
          firstName: { type: 'string' },
          lastName: { type: 'string' },
          email: { type: 'string' },
          type: { type: 'number' },
          pmi: { type: 'number' },
          timezone: { type: 'string' },
          verified: { type: 'number' },
          createdAt: { type: 'string' },
          lastLoginTime: { type: 'string' },
          lastClientVersion: { type: 'string' },
          picUrl: { type: 'string' },
          language: { type: 'string' },
          phoneNumber: { type: 'string' },
          status: { type: 'string' },
          roleId: { type: 'string' },
          dept: { type: 'string' },
          groupIds: {
            type: 'array',
            items: { type: 'string' },
          },
          hostKey: { type: 'string' },
        },
      },
    });

    const roles = context.jobState.collectedEntities.filter((e) =>
      e._type.includes('zoom_role'),
    );
    expect(roles.length).toBeGreaterThan(0);
    expect(roles).toMatchGraphObjectSchema({
      _class: ['AccessRole'],
      schema: {
        additionalProperties: false,
        properties: {
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          _type: { const: 'zoom_role' },
          id: { type: 'string' },
          name: { type: 'string' },
          totalMembers: { type: 'number' },
          description: { type: 'string' },
        },
      },
    });

    expect(
      context.jobState.collectedRelationships.filter(
        (e) => e._type === Relationships.GROUP_HAS_USER._type,
      ),
    ).toMatchDirectRelationshipSchema({
      schema: {
        properties: {
          _class: { const: 'HAS' },
          _type: {
            const: 'zoom_user_assigned_role',
          },
        },
      },
    });
  });
});
