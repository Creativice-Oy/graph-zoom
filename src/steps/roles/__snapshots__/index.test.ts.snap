// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchRoles should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "AccessRole",
      ],
      "_key": "zoom_role:0",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "description": "Account owner has full privileges to access and manage a Zoom account.",
            "id": "0",
            "name": "Owner",
            "total_members": 1,
          },
        },
      ],
      "_type": "zoom_role",
      "createdOn": undefined,
      "description": "Account owner has full privileges to access and manage a Zoom account.",
      "displayName": "Owner",
      "id": "0",
      "name": "Owner",
      "totalMembers": 1,
    },
    Object {
      "_class": Array [
        "AccessRole",
      ],
      "_key": "zoom_role:1",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "description": "Admins have wide range privileges to access and manage a Zoom account.",
            "id": "1",
            "name": "Admin",
            "total_members": 0,
          },
        },
      ],
      "_type": "zoom_role",
      "createdOn": undefined,
      "description": "Admins have wide range privileges to access and manage a Zoom account.",
      "displayName": "Admin",
      "id": "1",
      "name": "Admin",
      "totalMembers": 0,
    },
    Object {
      "_class": Array [
        "AccessRole",
      ],
      "_key": "zoom_role:2",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "description": "Members have access to basic Zoom video meeting functions but no account management privileges.",
            "id": "2",
            "name": "Member",
            "total_members": 1,
          },
        },
      ],
      "_type": "zoom_role",
      "createdOn": undefined,
      "description": "Members have access to basic Zoom video meeting functions but no account management privileges.",
      "displayName": "Member",
      "id": "2",
      "name": "Member",
      "totalMembers": 1,
    },
  ],
  "encounteredTypes": Array [
    "zoom_role",
  ],
  "numCollectedEntities": 3,
}
`;
